webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -334,7 +334,7 @@ private void writeLastContent(final Throwable throwable, final ChannelFutureList
     @Override
     public void onSubscribe(Flow.Subscription subscription) {
         this.subscription = subscription;
-        subscription.request(Long.MAX_VALUE);
+        subscription.request(2); // Because first chunk gets cached
```

Comment:
This is an incorrect assumption. I am afraid it is not correct in several cases, perhaps even in the case you are optimising for.

The first chunk gets cached conditionally. A safer approach is to request 1, and ensure 1 is requested for every invocation of `onNext` (ie guard all exits from that routine, except when deferred to `socket.write`).

Reply:
fixed